src/app/auth/callbackPage/CallbackPage.tsx

```diff
@@ -1,28 +1,45 @@
 import { useEffect } from 'react';
 import { RouteComponentProps } from 'react-router-dom';
+import { useMutation } from '@apollo/react-hooks';
 
 import LoadingPage from '../../../common/loadingPage/LoadingPage';
+import { CREATE_BERTH_PROFILE_MUTATION } from '../../../features/queries';
 import apolloClient from '../../apolloClient';
 import authService from '../authService';
 
 export type CallbackPageProps = RouteComponentProps;
 
 const CallbackPage = ({ history }: CallbackPageProps) => {
   const client = apolloClient;
+  const [createMyBerthProfile] = useMutation(CREATE_BERTH_PROFILE_MUTATION);
 
   useEffect(() => {
-    authService
-      .endLogin()
-      .then((user) => {
+    const endLogin = async () => {
+      try {
+        const user = await authService.endLogin();
+
+        // Create berth profile if it doesn't exist. If the profile does exist, this mutation
+        // returns the existing profile.
+        await createMyBerthProfile({
+          variables: {
+            input: {
+              profileToken: authService.getProfileToken(),
+            },
+          },
+        });
+
         client.writeData({
           data: { currentUser: { __typename: 'CurrentUser', ...user.profile } },
         });
+
         history.replace(user.state.path);
-      })
-      .catch(() => {
+      } catch (e) {
         // TODO: handle error
```

Comment:
I used `history.replace('/error')` here before the E2E test changes and it looked to be working when I tested it. However, I assume that there was a reason for removing it that didn't manifest in my testing.

Reply:
It was copied from the Admin UI, the issue here is that we don't have an error page :D so `/error` was leading to a `404` page.

Reply:
Ah :D My dev eyes couldn't separate that error from an error page